feat(user-operator): provide possibility to configure key name of use…

leisma opened the pull request:
### Type of change

- Enhancement / new feature

### Description

Adds support for configuring the certificate and key entry names in the Kubernetes Secret used by the standalone User Operator via the "STRIMZI_EO_KEY_NAME" environment variable. This allows the use of externally provided Secrets (not created by Strimzi) that contain certificate and key entries with names other than "entity-operator". If the environment variable is not set, the default value "entity-operator" is used, preserving the existing behavior.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

see-quick commented:
Hi, thanks for the contribution. Please sign your all commits to pass DCO :) [1]

[1] - https://github.com/strimzi/strimzi-kafka-operator/blob/main/CONTRIBUTING.md

scholzj reviewed (commented):
Assuming we want to support this:
* Should the User Operator have a default value instead of adding complexity to the Cluster Operator? That would also ensure backwards compatibility for existing tsandalone installations whcih would otherwise need to make sure they add this to the deployment files.
* Should we have separate config for public and private key? Because you come today with the problem that the _haerdcoded_ name does not work for you, which is fine. But tomorrow someone might have the problem that the same base name for priate and public key does not work for them.

leisma commented:
Thanks for the review!
I'm not sure if I understood all your points correctly, but I made the following changes:
- Removed the definitions from the Cluster Operator, so only the User Operator is affected by the change
- Added the ability to configure the names of the public and private keys separately

scholzj commented on documentation/modules/deploying/proc-deploy-user-operator-standalone.adoc:
```diff
@@ -147,10 +147,16 @@ env:
 ----
 <1> The Kubernetes `Secret` that contains the public key (`ca.crt`) value of the CA that signs Kafka broker certificates.
 <2> The Kubernetes `Secret` that contains the certificate public key (`entity-operator.crt`) and private key (`entity-operator.key`) that is used for mTLS authentication against the Kafka cluster.
+<3> The name of the private key that is used for mTLS authentication against the Kafka cluster. The default is `entity-operator`.
+<4> The name of the public key that is used for mTLS authentication against the Kafka cluster. The default is `entity-operator`.
```
Same as above ...

```suggestion
<3> The name of the private key that is used for mTLS authentication against the Kafka cluster. The default is `entity-operator.crt`.
<4> The name of the public key that is used for mTLS authentication against the Kafka cluster. The default is `entity-operator.key`.
```

This will likely affect the code part as well ...

leisma replied:
Same as above, did include that in my last commit.

scholzj reviewed (commented):
I think this goes in the direction I meant. But it is configuring the key underwhich it is stored in the Secret. So it should be directly the full key under which it is stored in the Secret and not something what is further modified?

leisma commented on documentation/modules/deploying/proc-deploy-user-operator-standalone.adoc:
```diff
@@ -147,10 +147,16 @@ env:
+  - name: STRIMZI_EO_KEY_NAME <3>
+    value: entity-operator
+  - name: STRIMZI_EO_CERT_NAME <4>
+    value: entity-operator
```
Thanks for the suggestion, I did include that in my last commit

scholzj commented:
/azp run regression

ppatierno commented on operator-common/src/main/java/io/strimzi/operator/common/auth/PemAuthIdentity.java:
```diff
@@ -60,19 +61,22 @@ private PemAuthIdentity(Secret secret, String secretCertName) {
     public static PemAuthIdentity clusterOperator(Secret secret) {
-        return new PemAuthIdentity(secret, "cluster-operator");
+        String privateAndPublicKeyName = "cluster-operator";
+        return new PemAuthIdentity(secret, String.format("%s.key", privateAndPublicKeyName), String.format("%s.crt", privateAndPublicKeyName));
```
the name of key and cert are pretty well defined, do we really need formatting a String?

scholzj replied:
I think there might be some method to do this already. But not at my desk right now, so I cannot find it. I can check later today or tomorrow.

leisma replied:
No not really, thought to use string formatting because of the dublication of the string "cluster-operator". Just fixed it and used the string directly. Thanks for the input!

scholzj replied:
I think the `SecretEntry` enum can be used for this maybe?

scholzj replied:
Anyway, works for me as it is right now if it works for others.

leisma replied:
@scholzj Sorry, I first did't notice your message and did the change. But for me the actual implementation works and is in my opinion quite easy to understand.

ppatierno reviewed (approved):
LGTM. Thanks!

scholzj reviewed (changes requested):
Some more nits I noticed a bit late, sorry. LGTM otherwise

scholzj commented:
/azp run regression
